pkg/mcp/testing/monitoring/reporter.go

```diff
@@ -99,3 +99,65 @@ func NewInMemoryServerStatsContext() *InMemoryServerStatsContext {
 		mutex:             &sync.Mutex{},
 	}
 }
+
+type nackKey struct {
+	error   string
+	typeURL string
+}
+
+type InMemoryClientStatsContext struct {
+	RequestAcksTotal   map[string]int64
+	RequestNacksTotal  map[nackKey]int64
+	SendFailuresTotal  map[errorCodeKey]int64
+	RecvFailuresTotal  map[errorCodeKey]int64
+	ReconnectionsTotal int64
+	mutex              *sync.Mutex
+}
```

Comment:
Could you define the mutex as a value and move to the top of the struct above the fields it protects? The zero value for a sync.Mutex is defined to be an unlocked mutex and doesn't require extra initialization. The standard grouping mechanism for defining mutex is to place it first rather than last.